### DISCUSSION TITLE
Optuna on XGBoost - should the optuna objective function returned score always alligned with xgboost goal ?

### DISCUSSION THREAD

**Discussion by pangjac** [optuna/optuna#5095] [Q&A] [answered]
I am referring this official [Optuna example](https://github.com/optuna/optuna-examples/blob/main/xgboost/xgboost_simple.py), where I noticed that the XGBoost model objective is `"objective": "binary:logistic",` and objective function is returning `accuracy = sklearn.metrics.accuracy_score(valid_y, pred_labels)`. This make senses and they are aligned with each other for a typical binary problem where the classes are balanced and the costs of false positives and false negatives are roughly equal.

My question is, should the Optuna objective function score always aligned with xgboost? 
For example,  assuming my dataset (fraud) is extremely imbalance, and the model metrics is not how many binary classification scores wrong, but how much money I lost. 

I have a customized evaluation metric defined something like 
```
def metrics_loss_money_total(valid_y, pred_lables):
    # some logic that calculate how much money loss if binary prediction error
    return total_money_loss
```
May I still use `"objective": "binary:logistic"` for XGBoost objective function, and use `metrics_loss_money_total` when returning the optuna objective function ?

**Comment by nzw0301** [comment, marked as answer]
> My question is, should the Optuna objective function score always aligned with xgboost?

This is not mandatory, but it allows us to optimise it effectively. 

I'm not sure about xgboost's objective functions, but using some cost-sensitive or imbalance-aware objectives rather logistic loss than is worth trying.
